api-v2/src/plan/plan.controller.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Patch,
+  Param,
+  Delete,
+  UseGuards,
+  Req,
+  BadRequestException,
+  ParseIntPipe,
+} from '@nestjs/common';
+import { PlanService } from './plan.service';
+import { CreatePlanDto } from './dto/create-plan.dto';
+import { UpdatePlanDto } from './dto/update-plan.dto';
+import { JwtAuthGuard } from 'src/guards/jwt-auth.guard';
+import { AuthenticatedRequest } from 'src/auth/interfaces/authenticated-request';
+import { Plan } from './entities/plan.entity';
+import { OwnPlanGuard } from 'src/guards/own-plan.guard';
+
+@Controller('plans')
+export class PlanController {
+  constructor(private readonly planService: PlanService) {}
+
+  /**
+   * Creates a plan for the signed in user.
+   */
+  @UseGuards(JwtAuthGuard)
+  @Post()
+  async create(
+    @Body() createPlanDto: CreatePlanDto,
+    @Req() req: AuthenticatedRequest,
+  ): Promise<Plan> {
+    try {
+      return await this.planService.create(createPlanDto, req.user);
+    } catch (error) {
+      throw new BadRequestException(error.message);
+    }
```

Comment:
save exceptions for exceptional behavior, in which case prefer 500. if it's valid for this to throw, then perhaps make its return type different

Reply:
Makes sense! also sorry, what do you mean by make its return type different? Do you mean make the return type of the service different(i.e undef if the creation failed and Plan if it didn't)?

Thought process behind this was that the `create` probably failed since the client request body had issues with table constraints we have. But you're right in that the create could also straight up fail for other 500 reasons(typeorm/db issues).

Reply:
yeah exactly. in general with JS/TS i prefer to not use try/catch because there aren't ways to match on what was thrown. for example, did we throw a normal `Error`, or a `PlanNotFound` error?

and yea sorry by "make return type different" i meant encode the possible failure modes into the type signature, so we can be sure that they are handled. this is nice, because then in the controller layer we can just throw the corresponding HttpError (or subclass). and then if something unexpected throws, it will get propagated automatically since we aren't try/catching everything. 

the only thing is there isn't a clean way to do this in typescript :((((

Reply:
discussed offline: decided to never throw in services (do try/catch)